packages/dso-toolkit/src/global/spacing.scss

```diff
@@ -64,3 +58,43 @@
     margin-block-start: units.$block-spacing-medium;
   }
 }
+
+$spacers: (
+  (
+    space: units.$block-spacing-small,
+    predecessors: (
+      ".dso-context-wrapper",
+    ),
+    successors: (
+      ".dso-table-responsive",
+    ),
+  ),
+  (
+    space: units.$block-spacing-medium,
+    predecessors: (
+      ".dso-app-heading",
+    ),
+    successors: (
+      ".dso-context-wrapper",
+    ),
+  ),
+  (
+    space: units.$block-spacing-xlarge,
+    predecessors: (
+      "dso-action-list",
+    ),
+    successors: (
+      ".dso-form-buttons",
+    ),
+  )
+);
+
+@each $spacer in $spacers {
+  $predecessors: map.get($spacer, predecessors);
+  $successors: map.get($spacer, successors);
+  $space: map.get($spacer, space);
+
+  :where(#{$predecessors}) + :where(#{$successors}, :has(#{$successors})) {
```

Comment:
Als je op deze plek de `:has()` toevoegt, gaat dat wel heel erg ver denk ik. Ik wil met `:has()` iets meer 'surgical' omgaan, dus alleen waar het waarschijnlijk is dat dit nodig is. We kunnen sowieso met een direct descendant het 'rabbit hole' probleem voorkomen: `:has(> #{$successors})`.

nu even wat `:hasj()` :)

Reply:
Ik denk ook dat het ver gaat, maar ik wil het graag _testen_ 😄

De direct descendant selector werkt in de basis hetzelfde, alleen niet recursief. Tekortkomingen aan de `:has()` oplossing zullen we ook bij `>` hebben, alleen is de kans kleiner.

Dus resume: Ergens denk ik dat dit niet kan werken, omdat het onbeperkt recursief gaat. Maar aan de andere kant denk ik dat het _wel_ kan werken, juist omdat we op een specifieke set testen.

Zie jij mogelijkheden om mijn voorzet te bevestigen of ontkrachten?

Reply:
In de voorbeeldpagina's kan je al aardig zien wat het doet. daar heb ik zelf mijn eigen oplossing ook getest.

 >> Ergens denk ik dat dit niet kan werken, omdat het onbeperkt recursief gaat
 Precies: de `>` voorkomt onbeperkt

>> Maar aan de andere kant denk ik dat het wel kan werken, juist omdat we op een specifieke set testen.
Ja, maar je wil ook dat het zo breed mogelijk werkt. een beetje een balanceer-act.

Reply:
Ik begrijp wat je zegt, maar zoals ik eerder zei:

> De direct descendant selector werkt in de basis hetzelfde, alleen niet recursief. Tekortkomingen aan de `:has()` oplossing zullen we ook bij `>` hebben, alleen is de kans (dan, _red_) kleiner.

Maar ik zal hier zelf nog wat testen op doen, laat het voor nu maar even rusten. Dank voor je input!